pkg/apis/controller/trials/v1alpha3/trial_types.go

```diff
@@ -36,10 +36,21 @@ type TrialSpec struct {
 	// Whether to retain the trial run object after completed.
 	RetainRun bool `json:"retainRun,omitempty"`
 
+	// Describes how objective value will be extracted from collected metrics
+	ObjectiveExtract ObjectiveExtractType `json:"objectiveExtract,omitempty"`
+
```

Comment:
Actually, we have an issue for this https://github.com/kubeflow/katib/issues/987. Maybe instead of set up another env variable, in each run we can have `min`, `max` and `latest` in Objective.

Reply:
/cc @gaocegege

Reply:
@andreyvelich The idea extracting min, max and latest objective in each run looks good to me. But I think we also need an additional field like `ObjectiveExtractType ` to describe which kind of objective chosen to be compared with goal.
Of course env variable `DefaultObjectiveExtract` is not necessary since the default choice can be figured out by ObjectiveType.

Reply:
Then, maybe it should be part of Experiment? 
What do you think @johnugeorge ?

Reply:
@andreyvelich @johnugeorge 
After a detailed review, I found the modification of  `common_types.Observation` will cause enormous changes including corresponding interface of all existing suggestion services. 
Maybe we can keep `suggestionapi.Observation` (pb spec)  unchanged ? Only change   `common_types.Observation` to 
```yaml
observation:
 metrics:
   - name: loss
     min: 0.0001
     max: 0.1234
     latest: 0.1111
```